gsl_projet/views.py

```diff
@@ -84,19 +87,58 @@ def get_context_data(self, **kwargs):
         return context
 
     def get_ordering(self):
-        sorting = self.request.GET.get("tri")
-        available_sortings = {
-            "date_de_depot": "dossier_ds__ds_date_depot",
-            "cout_total": "dossier_ds__finance_cout_total",
+        ordering_map = {
+            "date_desc": "-dossier_ds__ds_date_depot",
+            "date_asc": "dossier_ds__ds_date_depot",
+            "cout_desc": "-dossier_ds__finance_cout_total",
+            "cout_asc": "dossier_ds__finance_cout_total",
+            "commune_desc": "-address__commune__name",
+            "commune_asc": "address__commune__name",
         }
-        if sorting in available_sortings:
-            return available_sortings.get(sorting)
+
+        tri = self.request.GET.get("tri")
+        return ordering_map.get(tri, None)
 
     def get_queryset(self):
         qs = Projet.objects.for_user(self.request.user).filter(
             dossier_ds__ds_date_depot__gte=datetime.date(2024, 9, 1)
         )
+
+        # Filtre par dispositif
+        dispositif = self.request.GET.get("dispositif")
+        if dispositif:
+            qs = qs.filter(dossier_ds__demande_dispositif_sollicite=dispositif)
+
+        # Filtre par coût total minimum
+        cout_min = self.request.GET.get("cout_min")
+        if cout_min:
+            # qs = qs.filter(dossier_ds__finance_cout_total__gte=cout_min)
+            qs = qs.filter(
+                Q(assiette__isnull=False, assiette__gte=cout_min)
+                | Q(assiette__isnull=True, dossier_ds__finance_cout_total__gte=cout_min)
+            )
+
+        # Filtre par coût total maximum
+        cout_max = self.request.GET.get("cout_max")
+        if cout_max:
+            qs = qs.filter(
+                Q(assiette__isnull=False, assiette__lte=cout_max)
+                | Q(assiette__isnull=True, dossier_ds__finance_cout_total__lte=cout_max)
+            )
+
+        porteur = self.request.GET.get("porteur")
+        if porteur == "EPCI":
+            qs = qs.filter(
+                dossier_ds__porteur_de_projet_nature__label__in=NaturePorteurProjet.EPCI_NATURES
+            )
+        elif porteur == "Communes":
+            qs = qs.filter(
+                dossier_ds__porteur_de_projet_nature__label__in=NaturePorteurProjet.COMMUNE_NATURES
+            )
```

Comment:
Si on filtre un champ sur un ensemble limité de valeurs possibles, on peut imaginer un "mapping" comme tu as très justement fait pour les tris : 


```suggestion
        porteur_mappings = {
            "EPCI": NaturePorteurProjet.EPCI_NATURES,
            "Communes": NaturePorteurProjet.COMMUNE_NATURES,
        }
        porteur = self.request.GET.get("porteur")
        if porteur in porteur_mappings: # "in" vérifie la présence de la clé dans le dict
            qs = qs.filter(
                dossier_ds__porteur_de_projet_nature__label__in=porteur_mappings.get(porteur)
            )
```

Et le bonus c'est que tu as moins de copier-coller dans le template : tu peux passer ce mapping (ou juste ses clés) au contexte du template puis boucler dessus pour générer la liste des filtres.

Reply:
Bonne astuce !